Source/Core/Proj4Projection.js

```diff
@@ -0,0 +1,196 @@
+         * @type {String}
+         * @readonly
+         */
+        wellKnownText: {
+            get: function() {
+                return this._wkt;
+            }
+        },
+
+        /**
+         * The scale for converting from heights in meters to the projection's units.
+         * @memberof Proj4Projection.prototype
+         * @type {Number}
+         * @readonly
+         */
+        heightScale: {
+            get: function() {
+                return this._heightScale;
+            }
+        },
+
+        /**
+         * Gets whether or not the projection evenly maps meridians to vertical lines.
+         * Not all Proj4 projections are cylindrical about the equator.
+         *
+         * @memberof Proj4Projection.prototype
+         *
+         * @type {Boolean}
+         * @readonly
+         * @private
+         */
+        isNormalCylindrical : {
+            get : function() {
+                return false;
+            }
+        },
+
+        /**
+         * The bounds in Cartographic coordinates over which this projection is valid.
+         * @memberof Proj4Projection.prototype
+         * @type {Rectangle}
+         * @readonly
+         */
+        wgs84Bounds : {
+            get : function() {
+                return this._wgs84Bounds;
+            }
+        }
+    });
+
+    var scratchProjectionArray = [0.0, 0.0];
+    /**
+     * Projects a {@link Cartographic} coordinate, in radians, to map coordinates in meters based on
+     * the specified projection.
+     *
+     * @param {Cartographic} cartographic The coordinates to project.
+     * @param {Cartesian3} [result] An instance into which to copy the result.  If this parameter is
+     *        undefined, a new instance is created and returned.
+     * @returns {Cartesian3} The projected coordinates.  If the result parameter is not undefined, the
+     *          coordinates are copied there and that instance is returned.  Otherwise, a new instance is
+     *          created and returned.
+     */
+    Proj4Projection.prototype.project = function(cartographic, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('cartographic', cartographic);
+        //>>includeEnd('debug');
+
+        if (!defined(result)) {
+            result = new Cartesian3();
+        }
+
+        var wgs84Bounds = this.wgs84Bounds;
+        scratchProjectionArray[0] = CesiumMath.toDegrees(CesiumMath.clamp(cartographic.longitude, wgs84Bounds.west, wgs84Bounds.east));
+        scratchProjectionArray[1] = CesiumMath.toDegrees(CesiumMath.clamp(cartographic.latitude, wgs84Bounds.south, wgs84Bounds.north));
+
+        var projected;
+        try {
+            projected = this._projection.forward(scratchProjectionArray);
+        } catch(e) {
+            // Log a warning the first time a projection fails
+            oneTimeWarning('proj4js-project', 'proj4js forward failed for ' + scratchProjectionArray + ' with projection ' + this._wkt);
```

Comment:
Why is this exception just logged and ignored?  Does it happen as part of standard behavior?  That seems... bad.

Reply:
The failures halt rendering, and depending on the projection it can happen pretty frequently, especially if a limiter hasn't been applied. I figured it would be better to log something than make the projection completely unusable, I think I used to see this happen with EPSG 2039 and the occasional logs didn't introduce horribly noticeable jank.

I just noticed though that the camera for EPSG 2039 in Columbus View somehow broke since I last checked, so I'll take a look at that and try to add specs for this area.

Reply:
> used to see this happen with EPSG 2039

@mramato this was also with proj4js 2.4.4 though, and we're using 2.5.0 now so the same case from before might not apply anymore. So we could possibly take these checks out, I'd just be paranoid that it would turn up again and we'd have to re-add them.

Reply:
> I just noticed though that the camera for EPSG 2039 in Columbus View somehow broke since I last checked, so I'll take a look at that and try to add specs for this area.

It also got waaaaaay slower. Looks like some bounding spheres somewhere are getting recomputed very often which is extremely expensive now since we made the bounding spheres checks with weird projections do a bunch of projection.